design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,312 @@
+# Solution Design - GCP Authenticator
+
+## Table of Contents
+- [Glossary](#glossary)
+- [Useful links](#useful-links)
+- [Issue description](#issue-description)
+- [Out of scope](#out-of-scope)
+- [Solution](#solution)
+  * [GCP Resource Restrictions](#gcp-resource-restrictions)
+  * [Environment-provided service account google identity](#environment-provided-service-account-google-identity)
+  * [Access GCP authenticator](#access-gcp-authenticator)
+  * [Design](#design)
+    + [Class diagram](#class-diagram)
+    + [GCP authenticator flow](#gcp-authenticator-flow)
+  * [Backwards compatibility](#backwards-compatibility)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Open questions](#open-questions)
+- [Implementation plan](#implementation-plan)
+
+## Glossary
+
+| **Term** | **Description** |
+|----------|-----------------|
+|  GCP| Google Cloud Platform                 |
+|  Google compute engine (GCE)| Virtual machine in google (similar to AWS EC2)                  |
+|  Environment-provided service account| An identity for accessing private data on behalf of a service account inside Google Cloud environments |
+|  Service account key| App can self sign a google token with the service account key in order to accessing private data on behalf of a service account outside Google Cloud environments|
+
+
+
+## Useful links
+- [Feature doc](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+Authenticator#/)
+- [GCP authentication overview]( https://cloud.google.com/docs/authentication)
+- [Google metadata server](https://cloud.google.com/compute/docs/storing-retrieving-metadata#querying)
+
+## Issue description
+We want enable GCE hosts to authenticate Conjur with their google identity in order to retrieve secrets from Conjur.
+
+## Out of scope
+The only authentication strategy we are supporting is Environment-provided service account in google Compute engine.
+Other strategies are not supported: Service account key, OAuth 2.0 client, API key
+Other google components are not supported:  App Engine, GKE, Cloud Run, Cloud Functions
+
+## Solution
+We will add a new GCP authenticator to Conjur, which GCE hosts will can authenticate with by delivering their host Conjur identity and google identity as JWT token.
+
+As mentioned in the [feature doc](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+Authenticator#/), before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-gcp/mygcp
```

Comment:
please explain the logic behind this id

Reply:
This example is from the feature doc and its alligned with all other authneticators, see the comment above:
`conjur/authn-gcp/<service ID>` 
authn-gcp = name of authenticator
mygcp = service id

Reply:
`service id` is the GCP Service Accont ID?

Reply:
No, service id is for crating multiple authn gcp authenticators in conjur

Reply:
As long as the authenticator has no policy defined configuration what can drive a customer to define more than one GCP authenticator.
In other words can we suppress `service-id` for this authn method?

Reply:
That's a good question!! maybe we can have only one. @nessiLahav @eranha @Tovli wdyt? 
I also think we need to rename it to authn-gcp-metadata or authn-gcp-gce - wdyt?\

Reply:
@InbalZilberman i like the authn-gcp-gce 

> In other words can we suppress service-id for this authn method?

Good point i think we should suppress it in this use case
@InbalZilberman Update me if you are changing the feature doc